core/proto/keytransparency_v1_proto/keytransparency_v1_admin_proto.proto

```diff
@@ -20,30 +20,74 @@ package keytransparency.v1.proto;
+}
 
-  // app_id is the application to make this change for.
-  string app_id = 2;
+// GetAuthorizedKeysRequest is empty. 
+// TODO(gbelvin): Should we shard these keys by domain_id and app_id? 
+message GetAuthorizedKeysRequest{}
+
+// GetAuthorizedKeysResponse contains the set of public keys that correspond 
+// to the private keys the AccountRecoveryService has access to.
+message GetAuthorizedKeysResponse{
+  // AccountRecoveryKeys is a map from pub_key_id to the public key itself.
+  map<string, keyspb.PublicKey> account_recovery_keys = 1;
+  // ActiveKey is the key the service is currently using to sign it's operations.
+  string active_key_id = 2;
+}
 
-  // auth_pub_key_id is the id of the authorized_public key to use when updating accounts.
-  // This must be a key that this server has the private key for.
-  string auth_pub_key_id = 3;
+// CreateAccountRequest specifies the information with which a new account should be initialized.
+// New accounts will be signed with the current active key.
+// It is the responsibility of authorized callers to verify that domain_id/app_id/user_id is correct.
+message CreateAccountRequest {
+  // account is the account to create.
+  Account account = 1;
+  // add_account_recovery_keys specifies whether to add this service's account_recovery_keys to the set of authorized_keys.
+  // This must be set to true if any further operations from this API are meant to succeed. 
+  // If set to false, there must be at least one key in authorized_keys.
+  bool add_account_recovery_keys = 7;
 }
 
-// BatchUpdateEntriesResponse returns a list of users for which the set operation
-// was unsuccessful.
-message BatchUpdateEntriesResponse {
-  // errors is a map from user_ids to errors, if there was an error for that user.
-  map<string, string> errors = 1;
+// CreateAccountResponse represents success / failure. 
+message CreateAccountResponse{}
+
+// SetAccountDataRequest sets the data field for the account.
+// The account must have the service's current signing key in its list of authorized_keys in order to succeed.
+message SetAccountDataRequest {
+  // account contains data which will be applied to the account.
+  Account account = 1;
+}
+
+// SetAccountDataResponse indicates the sucess or failure of a SetAccountDataRequest.
+message SetAccountDataResponse {}
+
+// BatchCreateAccountRequest creates multiple accounts all at once.
+message BatchCreateAccountRequest {
+  repeated Account accounts = 1;
+  // add_account_recovery_keys specifies whether to add this service's account_recovery_keys to the set of authorized_keys.
+  // This must be set to true if any further operations from this API are meant to succeed. 
+  // If set to false, there must be at least one key in authorized_keys.
+  bool add_account_recovery_keys = 7;
 }
 
+message BatchCreateAccountResponse {
+  // errors is a list of errors in the same order as requests.
+  repeated google.rpc.Status errors = 1;
```

Comment:
Ditto for what goes in comes out. If the create input is `repeated Account`, I would expect that to be reflected in the output.

Since you also want the status code, and parallel lists are usually to be avoided, an out here is to have a `google.rpc.Status create_status` field in Account, which is never persisted and only ever set in BatchCreate's response.